src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java

```diff
@@ -0,0 +1,34 @@
+package net.sacredlabyrinth.phaed.simpleclans.events;
+
+import net.sacredlabyrinth.phaed.simpleclans.ClanPlayer;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * @author ThiagoROX
+ */
+public class ClanPlayerEvent extends PlayerEvent {
+
+    private static final HandlerList handlers = new HandlerList();
+    private final ClanPlayer clanPlayer;
+
+    public ClanPlayerEvent(@NotNull ClanPlayer clanPlayer) {
+        super(clanPlayer.toPlayer());
```

Comment:
I don't really like that clanPlayer.toPlayer() can throw NPE here, but I feel there is nothing to do with it.

Reply:
@RoinujNosde What do you think?

Reply:
Add Player as a constructor parameter.

Reply:
> Add Player as a constructor parameter.

It seems, it would break extended events.
So, breaking API change.